contracts/token/PermitExtension.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-FileCopyrightText: 2024 OpenZeppelin, Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.10;
+
+import {UnstructuredStorage} from "./UnstructuredStorage.sol";
+import {EIP712} from "@openzeppelin/contracts-v4.9/utils/cryptography/EIP712.sol";
```

Comment:
can use draft from 4.6.0 and implement `eip712Doman` as defined here: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.9.4/contracts/utils/cryptography/EIP712.sol#L117-L141
to support `IERC5267`